composer/callbacks/generate.py

```diff
@@ -41,7 +41,7 @@ def __init__(self,
         self.prompts = prompts
         self.generate_kwargs = kwargs
         self.batch_size = batch_size if batch_size is not None else len(prompts)
-        self.check_interval = create_interval_scheduler(interval, include_end_of_training=False)
+        self.check_interval = create_interval_scheduler(interval, include_end_of_training=True)
```

Comment:
this doesn't quite do what you want. Since this callback gets run on every event, any events that occur after the last batch is trained would end up triggering this. You need to add some extra logic to make sure it doesn't run more than once for the same batch or scope down which events the callback runs on. See the hf_checkpointer in foundry for the former solution.

Reply:
👍 Updated! And added tests